raster-foundry/raster-foundry
# Respect time and space filters to find scenes to update aoi projects

## Overview

This PR includes `created_at` and `acquisition_date` filters in the query for scenes for updating AOI
projects.

### Checklist

- [/] Any new SQL strings have tests (exercised in testing instructions, but we don't really have good batch project testing strategies I don't think)

## Notes

This is going to fail in CI, but it's actually fine to review. I'm working on a different batch subproject
thing for a minute to get to the "just fix the tests" stage, since fixing the tests is lower complexity
than the other thing.

## Testing Instructions
 
 * CI, plus...
 * rebuild your batch jar
 * run migrations
 * start up your server and create an AOI
 * kill your server, you don't need it anymore except your database and it costs resources
 * pick a scene and update its `created_at` and `acquisition_date` fields (I picked `a9ce69c2-f87e-4119-863d-d570afb53983`)
 * create an AOI with that scene's datasource (Landsat 8 if you're using the example) and draw an area that intersects the scene's bbox (the Solomon Islands, if you're using the example)   
 * run `rf update-aoi-project <your new project's id>`
 * confirm that it succeeds
 * confirm your scene is linked to the project in `scenes_to_projects`
 * confirm that `aois_last_checked` was updated on your project

Closes #3330
